User registration

FlisAnn opened the pull request:
[Api provides functionality to create user account] ( https://www.pivotaltracker.com/n/projects/2477606)
### User Story
```
As a restaurant API
In order to give visitors access to order functionality
I would like to be able to allow visitors to create an account
```

## Changes proposed in this pull request:
* Updating db migration with new tables and deletes some tables
* Setting up Factory bot
* Creating test for having a Factory
* Creating test for invalid emails
* Creating test for having DB columns
* Adding a new namespace
* Adding request test for registration functionality
* Configuring initializers

## What we have learned working on this feature:
* Tabbing to the left which makes us more efficient
* Know what you are doing and are going to do or you will have to redo it. 

##  Packages/Gems added
* gem 'devise_token_auth'

FlisAnn committed:
Adding devise_token_auth
Updating db migration with new tables and deletes some tables
Setting up Factory bot
Creating test for having a Factory
Creating test for invalid emails
Creating test for having db columns
Adding new namespace

FlisAnn committed:
Removing blank row

FlisAnn committed:
Adding request test for registration functionality
Configuring initializers

emtalen commented on app/models/user.rb:
```diff
@@ -0,0 +1,10 @@
+
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
```
You can remove these comments

emtalen commented on config/routes.rb:
```diff
@@ -1,5 +1,10 @@
 Rails.application.routes.draw do
+  mount_devise_token_auth_for 'User', at: 'api/v1/auth', skip: [:omniauth_callback]
```
Why are you using the namespace v1 for devise_token_auth but not for the products? Please change that to only use the namespace you decided for with the products

KyunginNa replied:
We were confused honestly...hahah...😂
Now we know what we are doing and because we've already made the client side to use api/products, we did as below to fix it :)
- Deleting namespace v1 from routes.rb
- Modifying urls of devise_token_auth and products
- Modifying urls in registrations_spec file
- Removing the folder v1 and adding the files in the api folder instead

emtalen reviewed (commented):
Also, please be a bit better at git ponging between each other so the commits not only come from one of you 👍🏻

FlisAnn committed:
Removing comments

Commit:
Deleted namespace v1 from routes.rb,
Modifying urls of devise_token_auth and products,
Modifying urls in registrations_spec file.

FlisAnn committed:
Removing the folder v1 and adding the files in the api filder instead
